domain/src/test/kotlin/no/nav/su/se/bakover/domain/BehandlingTest.kt

```diff
@@ -890,6 +892,174 @@ internal class BehandlingTest {
+            )
+            underkjent.opprettBeregning(saksbehandler, 1.januar(2020), 31.desember(2020))
+            underkjent.leggTilSimulering(saksbehandler, defaultSimulering())
+            underkjent.sendTilAttestering(saksbehandler)
+            underkjent.underkjenn(
+                Attestering.Underkjent(
+                    attestant = Attestant(navIdent = "1234"),
+                    grunn = Attestering.Underkjent.Grunn.ANDRE_FORHOLD,
+                    kommentar = "En fin kommentar om hvorfor denne innvilgelsen er feil."
+                )
+            )
+            underkjent.status() shouldBe UNDERKJENT_INNVILGET
+        }
+
+        @Test
+        fun `kan oppdatere behandlingsinformasjon`() {
+            underkjent.oppdaterBehandlingsinformasjon(
+                saksbehandler, extractBehandlingsinformasjon(underkjent).withVilkårAvslått()
+            )
+            underkjent.status() shouldBe VILKÅRSVURDERT_AVSLAG
+        }
+
+        @Test
+        fun `kan kjøre en ny beregning`() {
+            underkjent.opprettBeregning(
+                saksbehandler = saksbehandler,
+                fraOgMed = 1.januar(2021),
+                tilOgMed = 31.desember(2021),
+                fradrag = listOf()
+            )
+            underkjent.status() shouldBe BEREGNET_INNVILGET
+        }
+
+        @Test
+        fun `kan simulere`() {
+            underkjent.leggTilSimulering(saksbehandler, defaultSimulering())
+            underkjent.status() shouldBe SIMULERT
+        }
+
+        @Test
+        fun `kan sende til attestering`() {
+            underkjent.sendTilAttestering(saksbehandler)
+            underkjent.status() shouldBe TIL_ATTESTERING_INNVILGET
+        }
+    }
+
+    @Nested
+    inner class UnderkjentAvslag_vilkårsvurderinger {
+        private lateinit var underkjent: Behandling
+
+        @BeforeEach
+        fun beforeEach() {
+            underkjent = createBehandling(id1, OPPRETTET)
+            underkjent.oppdaterBehandlingsinformasjon(
+                saksbehandler,
+                extractBehandlingsinformasjon(underkjent).withVilkårAvslått()
+            )
+            underkjent.sendTilAttestering(saksbehandler)
+            underkjent.underkjenn(
+                Attestering.Underkjent(
+                    attestant = Attestant(navIdent = "1234"),
+                    grunn = Attestering.Underkjent.Grunn.INNGANGSVILKÅRENE_ER_FEILVURDERT,
+                    kommentar = "En ufin kommentar om hvorfor dette avslaget er feil"
+                )
+            )
+            underkjent.status() shouldBe UNDERKJENT_AVSLAG
+        }
+
+        @Test
+        fun `kan oppdatere behandlingsinformasjon`() {
+            underkjent.oppdaterBehandlingsinformasjon(
+                saksbehandler,
+                extractBehandlingsinformasjon(underkjent).withAlleVilkårOppfylt()
+            )
+            underkjent.status() shouldBe VILKÅRSVURDERT_INNVILGET
```

Comment:
Kan vel og komme til VILKÅRSVURDERT_AVLSAG? OPPRETTET?

Reply:
VILKÅRSVURDERT_AVLSAG blir testet i kan oppdatere behandlingsinformasjon().

Har lagt til for OPPRETTET her, og over.